python/cudaq/kernel/kernel_decorator.py

```diff
@@ -184,7 +184,11 @@ def compile(self):
                 }
                 if self.dependentCaptures != None:
                     for k, v in self.dependentCaptures.items():
-                        if self.globalScopedVars[k] != v:
+                        if (isinstance(v, (list, np.ndarray))):
+                            # We can't detect a change in an array, always recompile.
+                            self.module = None
```

Comment:
If one were running kernels doing `cudaq.vqe()` or with `optimzer.optimize`, does this mean it would have to recompile every time through the optimization loop? If so, that might have a negative performance impact.

Reply:
I think so. I think the problem is that we don't save the list argument to compare to... Let me see if I can just copy it into a kernel decorator for comparison, so I can compare the list elements and skip the recompilation. The comparison would have to happen every time, which less than recompilation. I want to eventually avoid that too (https://github.com/NVIDIA/cuda-quantum/issues/1929) but for now just would like to fix the bug. Will update.

Reply:
Done!